shopinvader_delivery_carrier/tests/test_carrier.py

```diff
@@ -71,3 +79,210 @@ def test_reset_carrier_on_delte_item(self):
+        cart_values_after = self.cart._convert_to_write(self.cart._cache)
+        cart_values_after.pop("order_line", None)
+        self.assertDictEqual(cart_values_before, cart_values_after)
+        nb_lines_after = self.env["sale.order.line"].search_count([])
+        self.assertEquals(nb_lines_after, nb_lines_before)
+        # Ensure lines still ok
+        self.assertEquals(len(lines), len(self.cart.order_line))
+        for line_id, line_values in lines.iteritems():
+            order_line = self.cart.order_line.filtered(
+                lambda l, lid=line_id: l.id == lid
+            )
+            # Because delivery line has changed and the ID doesn't match
+            # anymore.
+            # But should still similar!
+            if not order_line:
+                order_line = self.cart.order_line.filtered(
+                    lambda l: l.is_delivery
+                )
+            order_line.read()
+            self.assertDictEqual(
+                order_line._convert_to_write(order_line._cache), line_values
+            )
+        self.assertEquals(self.cart.partner_id, partner)
+        self.assertEquals(french_country, partner.country_id)
+        self._check_carriers(result)
+
+    def test_get_cart_price_by_country3(self):
+        """
+        Check the service get_cart_price_by_country.
+        For this case, the cart have 1 delivery line set
+        The cart doesn't have a carrier set.
+        :return:
+        """
+        french_country = self.env.ref("base.fr")
+        belgium = self.env.ref("base.be")
+        partner = self.cart.partner_id
+        partner.write({"country_id": french_country.id})
+        self.assertFalse(self.cart.carrier_id)
+        # Force load every fields
+        self.cart.read()
+        cart_values_before = self.cart._convert_to_write(self.cart._cache)
+        cart_values_before.pop("order_line", None)
+        lines = {}
+        for line in self.cart.order_line:
+            line.read()
+            lines.update({line.id: line._convert_to_write(line._cache)})
+        nb_lines_before = self.env["sale.order.line"].search_count([])
+        self.service.shopinvader_session.update({"cart_id": self.cart.id})
+        params = {"country_id": belgium.id, "target": "current_cart"}
+        result = self.carrier_service.dispatch("search", params=params)
+        self.assertEquals(self.cart.name, cart_values_before.get("name", ""))
+        self.cart.read()
+        cart_values_after = self.cart._convert_to_write(self.cart._cache)
+        cart_values_after.pop("order_line", None)
+        self.assertDictEqual(cart_values_before, cart_values_after)
+        nb_lines_after = self.env["sale.order.line"].search_count([])
+        self.assertEquals(nb_lines_after, nb_lines_before)
+        # Ensure lines still ok
+        self.assertEquals(len(lines), len(self.cart.order_line))
+        for line_id, line_values in lines.iteritems():
+            order_line = self.cart.order_line.filtered(
```

Comment:
I don't get this: shouldn't you simply check that the old line is not there anymore and that the new line has the values coming from the new country shipping?

Reply:
I remove my next line (`if not order_line:`) because it's always the same order_line. This use case has been updated.